File: LeetCode-Programs/Sort Array By Parity II.py
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        s=len(nums)
        i,j=0,1
        while i<s and j<s:
            if nums[i]%2==0:
                i=i+2
            if nums[j]%2==1:
                j=j+2
            else:
                nums[i],nums[j] = nums[j],nums[i]
        return nums

File: LeetCode-Programs/Sort Array By Parity II.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        int s=nums.size(), i=0, j=1;
        while(i<s && j<s)
        {
            if(nums[i]%2==0)
                i=i+2;
            else if(nums[j]%2==1)
                j=j+2;
            else
                swap(nums[i], nums[j]);
        }
        return nums;
    }
};
